TESTS/mbedtls/config/aes_test.cpp

```diff
@@ -0,0 +1,547 @@
+    unsigned char buf[64];
+    mbedtls_aes_context ctx;
+    memset(key, 0, 32);
+    mbedtls_aes_init(&ctx);
+    const unsigned char *aes_tests;
+    for (int i = 0; i < 6; i++) {
+        u = i >> 1;
+        keybits = 128 + u * 64;
+        mode = i & 1;
+
+        if (verbose != 0)
+            printf("  AES-ECB-%3d (%s): ", keybits,
+                   (mode == MBEDTLS_AES_DECRYPT) ? "dec" : "enc");
+
+        memset(buf, 0, 16);
+
+        if (mode == MBEDTLS_AES_DECRYPT) {
+            ret = mbedtls_aes_setkey_dec(&ctx, key, keybits);
+            aes_tests = aes_test_ecb_dec[u];
+        } else {
+            ret = mbedtls_aes_setkey_enc(&ctx, key, keybits);
+            aes_tests = aes_test_ecb_enc[u];
+        }
+
+        /*
+         * AES-192 is an optional feature that may be unavailable when
+         * there is an alternative underlying implementation i.e. when
+         * MBEDTLS_AES_ALT is defined.
+         */
+        if (ret == MBEDTLS_ERR_AES_FEATURE_UNAVAILABLE && keybits == 192) {
+            printf("skipped\n");
+            continue;
+        } else if (ret != 0) {
+            goto exit;
+        }
+
+        for (int j = 0; j < 10000; j++) {
+            ret = mbedtls_aes_crypt_ecb(&ctx, mode, buf, buf);
+            if (ret != 0) {
+                goto exit;
+            }
+        }
+
+        if (memcmp(buf, aes_tests, 16) != 0) {
+            ret = 1;
+            goto exit;
+        }
+
+        if (verbose != 0) {
+            printf("passed\n");
+        }
+    }
+
+    if (verbose != 0) {
+        printf("\n");
+    }
+exit:
+    mbedtls_aes_free(&ctx);
+    return ret;
+}
+
+#if defined(MBEDTLS_CIPHER_MODE_CBC)
+int mbedtls_aes_cbc_test_mbedos(int verbose)
+{
+    int ret;
+    int u;
+    int mode;
+    unsigned int keybits;
+    unsigned char key[32];
+    unsigned char buf[64];
+    unsigned char iv[16];
+    unsigned char prv[16];
+    mbedtls_aes_context ctx;
+    memset(key, 0, 32);
+    mbedtls_aes_init(&ctx);
+    const unsigned char *aes_tests;
+    /*
+     * CBC mode
+     */
+    for (int i = 0; i < 6; i++) {
+        u = i >> 1;
+        keybits = 128 + u * 64;
+        mode = i & 1;
+
+        if (verbose != 0)
+            printf("  AES-CBC-%3d (%s): ", keybits,
+                   (mode == MBEDTLS_AES_DECRYPT) ? "dec" : "enc");
+
+        memset(iv, 0, 16);
+        memset(prv, 0, 16);
+        memset(buf, 0, 16);
+
+        if (mode == MBEDTLS_AES_DECRYPT) {
+            ret = mbedtls_aes_setkey_dec(&ctx, key, keybits);
+            aes_tests = aes_test_cbc_dec[u];
+        } else {
+            ret = mbedtls_aes_setkey_enc(&ctx, key, keybits);
+            aes_tests = aes_test_cbc_enc[u];
+        }
+
+        /*
+         * AES-192 is an optional feature that may be unavailable when
+         * there is an alternative underlying implementation i.e. when
+         * MBEDTLS_AES_ALT is defined.
+         */
+        if (ret == MBEDTLS_ERR_AES_FEATURE_UNAVAILABLE && keybits == 192) {
```

Comment:
The error code should be `MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED` for every key size, see https://github.com/ARMmbed/mbedtls/pull/2124/files#diff-b621d92c71bafecc8ae442a2bff4d1fdR1980

Reply:
will change